### PR TITLE
Remove ICode

### PR DESCRIPTION
**Pull request by soc**

### CODE REVIEW

**Comment by soc** [comment]
Two tests should be failing, commit message will be added, etc.
Compiling here, because I still have that annoying hanging test case locally.

File: src/compiler/scala/tools/nsc/Global.scala [hunk 1]
```diff
@@ -1415,9 +1334,8 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
       inform(fmt format toCheck.name)
 
       if (canCheck) {
-        phase = globalPhase
-        if (globalPhase.id >= icodePhase.id) icodeChecker.checkICodes()
```
Comment (soc) [inline src/compiler/scala/tools/nsc/Global.scala:1419]: I believe that this hack could be related to the failure seen in pos/z1730

**Comment by soc** [comment]
Review by @lrytz?

**Comment by lrytz** [comment]
Closing in favor of #4838

**Pull request closed without merging**